src/uct/api/uct.h

```diff
@@ -2108,16 +2112,17 @@ UCT_INLINE_API ucs_status_t uct_ep_tag_eager_short(uct_ep_h ep, uct_tag_t tag,
  *                        receiver.
  * @param [in]  pack_cb   User callback to pack the data.
  * @param [in]  arg       Custom argument to @a pack_cb.
+ * @param [in]  flags     Tag message flags, see @ref uct_msg_flags.
```

Comment:
Shouldn't `imm` be paired with corresponding flag? what if user wants to pass `0`-value as `imm`? The same for `zcopy`.

Reply:
0 can be passed as imm value as well. Any value can be passed as immediate. But in IB we have optimization that we do not send immediate data if it is 0. Then, when receiver gets regular SEND without  IB verbs _WITH_IMM flag, it means that immediate value is 0. 
I'd not introduce new flag to avoid complicating the API.

Reply:
ok for now, but what if we want add `put_with_imm` in future? API will not be aligned. @yosefe maybe we have to think about it for 2.0?

Reply:
what is the problem with put_with_imm? we will just add "immediate value" param, but not flags, since it's one-sided operation.

Reply:
No problem if it will be separated function